src/Formik.tsx

```diff
@@ -215,23 +215,29 @@ export class Formik<Values = object, ExtraProps = {}> extends React.Component<
           }
           return prev;
         },
-        {} as FormikErrors<Values>
+        {} as Error | FormikErrors<Values>
       )
     );
   }
 
-  runValidateHandler(values: FormikValues): Promise<FormikErrors<Values>> {
-    return new Promise(resolve => {
+  runValidateHandler(
+    values: FormikValues
+  ): Promise<Error | FormikErrors<Values>> {
+    return new Promise((resolve, reject) => {
       const maybePromisedErrors = (this.props.validate as any)(values);
       if (maybePromisedErrors === undefined) {
         resolve({});
       } else if (isPromise(maybePromisedErrors)) {
         (maybePromisedErrors as Promise<any>).then(
-          () => {
-            resolve({});
+          maybeErrors => {
+            resolve(maybeErrors);
           },
           errors => {
-            resolve(errors);
+            if (errors instanceof Error) {
```

Comment:
In this case, an error was thrown within the promise in the `validate` function.


```js
const validate = values => new Promise(resolve => {
  const errors = {};
  if (!values.name) errors.name = "Missing required value";
  throw new Error('Oops');
  resolve(errors);
});
```

We need to **reject** this promise, otherwise the `deepmerge.all` in the `runValidations` will swallow the error.

https://github.com/jaredpalmer/formik/blob/13954fdd10070f727a0259dbbf98824cb980fa92/src/Formik.tsx#L273-L284

<img width="365" alt="image" src="https://user-images.githubusercontent.com/1110551/50241970-4310e000-03c9-11e9-95ad-53db9d859d9f.png">